examples/2-advanced/variable-depth.ipynb

```diff
@@ -0,0 +1,521 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Variable Depth"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import itertools\n",
+    "\n",
+    "import healpy as hp\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "import scipy.interpolate\n",
+    "\n",
+    "# use the CAMB cosmology that generated the matter power spectra\n",
+    "import camb\n",
+    "from cosmology import Cosmology\n",
+    "\n",
+    "# almost all GLASS functionality is available from the `glass` namespace\n",
+    "import glass\n",
+    "import glass.ext.camb\n",
+    "\n",
+    "# creating a numpy random number generator for sampling\n",
+    "rng = np.random.default_rng(seed=42)\n",
+    "\n",
+    "# cosmology for the simulation\n",
+    "h = 0.7\n",
+    "Oc = 0.25\n",
+    "Ob = 0.05\n",
+    "\n",
+    "# basic parameters of the simulation\n",
+    "nside = lmax = 64\n",
```

Comment:
The resolution was initially 1024. I have had to drop it to 64 as otherwise this notebook takes a really long time. 128 would be suitable, but the pre-allocation array step becomes too big in CI (based on GH runners), so went with 64. Still gives similar results (by eye).